app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,74 @@
+class TweeetsController < ApplicationController
+  get '/tweets/new' do
+    if !session[:user_id]
+      redirect '/login'
+    else
+      erb :'/tweets/create'
+    end
+  end
+  get '/tweets' do
+    @tweets = Tweet.all
+    if !session[:user_id]
+      redirect '/login'
+    else
+      erb :'/tweets/show'
+    end
+  end
+  get '/tweets/:id' do
+    if session[:user_id]
+      @tweet = Tweet.find_by_id(params[:id])
+      erb :'tweets/tweet'
+    else
+      redirect to '/login'
+    end
+  end
+  get '/tweets/:id/edit' do
+    if session[:user_id]
+      @tweet = Tweet.find_by_id(params[:id])
+      if @tweet.user_id == session[:user_id]
+       erb :'/tweets/edit'
+      else
+        redirect to '/tweets'
+      end
+    else
+      redirect to '/login'
+    end
+  end
+
+  patch '/tweets/:id' do
+    if params[:content] == ""
+      redirect to "/tweets/#{params[:id]}/edit"
+    else
+      @tweet = Tweet.find_by_id(params[:id])
```

Comment:
we don't need to use an instance variable here on line 42 as we are not going to be using this in the view. 
I would set line 43 and 44 up using the Active Record update method 

```ruby 
tweet.update(content: params[:content]
``` 
and now we won't need to use the `.save` method anymore